components/brave_ads/core/internal/user_engagement/ad_events/search_result_ads/search_result_ad_event_handler.cc

```diff
@@ -50,28 +64,40 @@ void SearchResultAdEventHandler::FireEvent(
     BLOG(1,
          "Failed to fire search result ad event due to the ad being invalid");
 
-    return FailedToFireEvent(ad, event_type, std::move(callback));
+    return FailedToFireEvent(
+        ad, event_type,
+        base::BindOnce(&CompletedCallback, std::move(mojom_creative_ad),
+                       std::move(callback)));
   }
 
-  if (event_type == mojom::SearchResultAdEventType::kServedImpression &&
-      !SearchResultAdPermissionRules::HasPermission()) {
-    BLOG(1, "Search result ad: Not allowed due to permission rules");
-    return FailedToFireEvent(ad, event_type, std::move(callback));
+  if (!CanFireEvent(mojom_creative_ad, event_type)) {
+    return FailedToFireEvent(
+        ad, event_type,
+        base::BindOnce(&CompletedCallback, std::move(mojom_creative_ad),
+                       std::move(callback)));
   }
 
   switch (event_type) {
     case mojom::SearchResultAdEventType::kServedImpression: {
-      FireEvent(ad, event_type, std::move(callback));
+      MaybeFireServedEvent(
+          ad, base::BindOnce(&CompletedCallback, std::move(mojom_creative_ad),
+                             std::move(callback)));
       break;
     }
 
     case mojom::SearchResultAdEventType::kViewedImpression: {
-      FireViewedEvent(std::move(mojom_creative_ad), std::move(callback));
+      MaybeFireViewedEvent(
+          ad, FromMojomBuildDeposit(mojom_creative_ad),
+          base::BindOnce(&CompletedCallback, std::move(mojom_creative_ad),
+                         std::move(callback)));
       break;
     }
 
     case mojom::SearchResultAdEventType::kClicked: {
-      FireClickedEvent(ad, std::move(callback));
+      MaybeFireEvent(
```

Comment:
This is only called for clicks, maybe we can rename it for consistency with other funcntions?

Reply:
This is also called from `MaybeFireViewedEventCallback`, thanks